### ISSUE TITLE
Slowness in UI rendering after Samsung Security update (1st August 2023)

### ISSUE THREAD

**Issue by PradeepKK13** [facebook/react-native#39295]
### Description

Our application is experiencing a slowdown in rendering the user interface and handling state changes when multiple API calls occur simultaneously on Samsung devices, particularly those updated with the new security patch released in August 2023. However, it continues to function smoothly on other Samsung devices that have the older security patch from May 2023. We utilize the Axios package for making API requests. May I request an analysis of the possible root causes contributing to this inconsistency?

### React Native Version

0.72.4

### Output of `npx react-native info`

System:

  OS: macOS 13.5.1

  CPU: (10) arm64 Apple M1 Pro

  Memory: 123.80 MB / 16.00 GB

  Shell:

    version: "5.9"

    path: /bin/zsh

Binaries:

  Node:

    version: 20.5.1

    path: /opt/homebrew/bin/node

  Yarn:

    version: 1.22.19

    path: /usr/local/bin/yarn

  npm:

    version: 9.8.1

    path: /opt/homebrew/bin/npm

  Watchman:

    version: 2023.08.14.00

    path: /opt/homebrew/bin/watchman

Managers:

  CocoaPods:

    version: 1.12.1

    path: /opt/homebrew/bin/pod

SDKs:

  iOS SDK:

    Platforms:

      - DriverKit 22.4

      - iOS 16.4

      - macOS 13.3

      - tvOS 16.4

      - watchOS 9.4

  Android SDK: Not Found

IDEs:

  Android Studio: 2022.3 AI-223.8836.35.2231.10671973

  Xcode:

    version: 14.3.1/14E300c

    path: /usr/bin/xcodebuild

Languages:

  Java:

    version: 11.0.20

    path: /usr/bin/javac

  Ruby:

    version: 2.6.10

    path: /usr/bin/ruby

npmPackages:

  "@react-native-community/cli": Not Found

  react:

    installed: 18.2.0

    wanted: 18.2.0

  react-native:

    installed: 0.72.4

    wanted: 0.72.4

  react-native-macos: Not Found

npmGlobalPackages:

  "*react-native*": Not Found

Android:

  hermesEnabled: true

  newArchEnabled: false

iOS:

  hermesEnabled: true

  newArchEnabled: false

### Steps to reproduce

It is happening when there are multiple API calls in parallel.

### Snack, screenshot, or link to a repository

![MicrosoftTeams-image (97)](https://github.com/facebook/react-native/assets/51114651/8a758425-84d4-4557-bafa-fa5d17875dbb)

**Comment by cortinico** [comment]
Closing as a duplicate of #39183
